packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/WebChromeClientHostApiImpl.java

```diff
@@ -38,6 +40,8 @@ public static class WebChromeClientImpl extends SecureWebChromeClient {
     private final WebChromeClientFlutterApiImpl flutterApi;
     private boolean returnValueForOnShowFileChooser = false;
 
+    private boolean hasJavaScriptDialogCallback = false;
```

Comment:
Since the platform interface separates handling the callbacks into three separate methods, you should also have three separate flags for each `alert`/`confirm`/`prompt`. What happens if a user only wants to set the `confirm` dialog?